src/concrete/ml/quantization/quantized_module.py

```diff
@@ -337,26 +337,31 @@ def forward(
         q_x = to_tuple(self.quantize_input(*x))
 
         if debug and fhe == "disable":
-            debug_value_tracker: Optional[
-                Dict[str, Dict[Union[int, str], Optional[ONNXOpInputOutputType]]]
+            debug_value_tracker: Dict[
+                str, Dict[Union[int, str], Optional[ONNXOpInputOutputType]]
             ] = {}
             for (_, layer) in self.quant_layers_dict.values():
                 layer.debug_value_tracker = debug_value_tracker
-            result = self.quantized_forward(*q_x, fhe="disable")
+            q_y_pred = self.quantized_forward(*q_x, fhe="disable")
             for (_, layer) in self.quant_layers_dict.values():
                 layer.debug_value_tracker = None
-            return result, debug_value_tracker
+            # De-quantize the output predicted values
+            y_pred = self.dequantize_output(
+                *([q_y_pred] if isinstance(q_y_pred, numpy.ndarray) else q_y_pred)
+            )
+            return y_pred, debug_value_tracker
 
         q_y_pred = self.quantized_forward(*q_x, fhe=fhe)
 
         # De-quantize the output predicted values
-        y_pred = self.dequantize_output(q_y_pred)
-
+        y_pred = self.dequantize_output(
```

Comment:
this seems quite impractical and I feel we could have a nicer way of doing it ! But at the same time, I'm not sure what could be the best, the others possibilities would be : 
- make `quantized_forward` return a tuple, even with a single value, and make `dequantize_output` only handle tuple of arrrays as input
- keep `quantized_forward` like you did, but then make `dequantize_output` handle numpy array or tuple of numpy array as input, and handle that and use the `to_tuple` utility function that we have

I think I prefer the last option as it's already something we do in the `forward` for example (since `quantize_inputs` have this similar behavior of removing the tuple type if there's a single value

worse case you might want to use `to_tuple` here instead of this custom if/else statement !